src/kernels/kernel.ts

```diff
@@ -277,11 +276,20 @@ abstract class BaseKernel implements IBaseKernel {
                 Array.from(this.hooks.get('willRestart') || new Set<Hook>()).map((h) => h(this._jupyterSessionPromise))
             );
             traceInfo(`Restart requested ${this.uri}`);
-            this.startCancellation.cancel();
+            this.startCancellation.cancel(); // Cancel any pending starts.
```

Comment:
This alone will not work, as the previous code line 280 was used to cancel the pending start, hence cancelling the previous and immediately initializing it with a new item will not have the desired effect of the cancellation.

Personally not sure why we're cancellign either, but this is existing code and reluctant to change that at this stage, will look into that in next iteration debt.

Reply:
initializing a new `cancellationSource` won't affect the token that was previously passed to the tasks that would need to be cancelled.